### PR TITLE
feat(ai): migrate localNlpService onto WorkerBus v2, delete v1 inference worker

### PR DESCRIPTION
**Pull request by qnbs**
## Summary
Fourth and final migration PR of the worker-generation consolidation ([ADR-0014](docs/adr/0014-worker-generation-duplication.md)). Stacked on #288 → #287 → #286. Plan: `.claude/plans/worker-generation-v1-vs-enumerated-fox.md`.

- `services/ai/localNlpService.ts`'s public API (`analyzeSentiment`/`summarizeText`/`classifyWritingTopic`) is **unchanged** — confirmed zero real production callers beyond its own tests and a barrel re-export in `services/ai/index.ts`, so this is the lowest-blast-radius of the three service migrations. Internals swapped from a dedicated `new Worker(inference.worker.ts)` instance to the shared WorkerBus v2 `'inference'` pool via `ensureInferencePool()` (added in #288), routing to the `'inference.text'` capability.
- `analyzeSentiment`/`summarizeText` already had their own app-level graceful-degrade behavior on failure (NEUTRAL sentiment / truncated text) — preserved unchanged via try/catch around the new throw-on-failure `requestInference()` call.
- **With both `localEmbeddingService.ts` (#288) and this migrated, `workers/inference.worker.ts` (v1) has no remaining callers — deleted**, along with its dedicated test (`tests/unit/inferenceWorker.test.ts`).
- `tests/unit/localNlpService.test.ts` rewritten against a `workerBusManager` mock; `tests/unit/services/localNlpService.test.ts` deleted as a near-total duplicate (same module, overlapping `classifyWritingTopic`/`analyzeSentiment` coverage the rewritten suite already has).

Verified via typecheck **and** a full local `pnpm run build:edge` reproduction of Vercel's exact build command — the lesson from #288's missing-export incident (typecheck/Vitest alone didn't catch that one; only the real build did).

## Test plan
- [x] `pnpm exec vitest run tests/unit/localNlpService.test.ts tests/unit/inferenceWorkerHandlerV2.test.ts` — 30/30 passing
- [x] `pnpm run typecheck` (exact CI command) — clean
- [x] `NODE_OPTIONS=--max-old-space-size=3072 pnpm run build:edge` (exact Vercel build command) — clean, verified end-to-end
- [x] `pnpm run lint` (Biome) — clean via pre-commit hook
- [ ] CI quality gate green

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Improvements**
  * Updated local sentiment analysis and text summarization to run on the shared inference system (WorkerBus v2).
  * Strengthened resiliency by falling back to safe defaults on any inference or parsing failure.
  * Continued enforcing input truncation and summarization limits/options.
* **Tests**
  * Updated NLP service unit tests to mock the shared inference/queue flow.
  * Removed obsolete unit tests related to the retired message-based inference worker.
* **Documentation**
  * Refreshed cache header notes for the inference pipeline LRU cache.
* **Chores**
  * Minor typing-style comment update in the WASM speech-to-text engine.

### CODE REVIEW

File: services/ai/localNlpService.ts [hunk 1]
```diff
@@ -12,74 +12,59 @@ const SENTIMENT_MODEL = 'Xenova/distilbert-base-uncased-finetuned-sst-2-english'
...
+  const handle = bus.enqueue<
+    { task: string; modelId: string; input: string; inferenceOptions?: Record<string, unknown> },
+    string
+  >(
+    'inference.text',
+    { task, modelId, input, inferenceOptions },
+    { capabilities: ['inference.text'] },
+  );
+  return handle.result;
 }
```
[the remark this answers is not in the document]
Comment (qnbs) [reply services/ai/localNlpService.ts:36]: Confirmed and fixed in fed9b28b — `requestInference()` was spreading `inferenceOptions` into the enqueue payload via object-literal shorthand, so calls without a 4th arg (e.g. `analyzeSentiment`) set the key to an explicit `undefined`. With `exactOptionalPropertyTypes: true` the optional payload field only accepts the property being *absent*, not present-with-undefined. Fixed by conditionally spreading the key in only when `inferenceOptions !== undefined`, and added a regression test asserting the key is omitted (`Object.hasOwn(...) === false`).

File: tests/unit/localNlpService.test.ts [hunk 2]
```diff
@@ -1,163 +1,152 @@
...
+      'inference.text',
+      expect.anything(),
+      expect.objectContaining({ capabilities: ['inference.text'] }),
+    );
+  });
+
+  it('omits the inferenceOptions key entirely when none is passed (exactOptionalPropertyTypes)', async () => {
+    await analyzeSentiment('test');
+    expect(Object.hasOwn(requestCalls[0] ?? {}, 'inferenceOptions')).toBe(false);
+  });
```
[the remark this answers is not in the document]
Comment (qnbs) [reply tests/unit/localNlpService.test.ts:120]: Fixed in cfb0fc0e — added `expect(requestCalls).toHaveLength(1)` before the hasOwn check.

File: tests/unit/localNlpService.test.ts [hunk 3]
```diff
@@ -1,163 +1,152 @@
...
     expect(r.normalized).toBe(0);
   });
+
+  it('defaults score to 0.5 when the worker omits the ":score" suffix', async () => {
+    nextResult = 'POSITIVE';
+    const r = await analyzeSentiment('no score suffix');
+    expect(r.label).toBe('POSITIVE');
+    expect(r.score).toBe(0.5);
+  });
+
```
[the remark this answers is not in the document]
Comment (qnbs) [reply tests/unit/localNlpService.test.ts:105]: Not adding this — checked all 8 tests in this describe('analyzeSentiment') block and none have individual QNBS-v3 comments, only the file-level one at the top of the file. Adding one to just these 2 new tests would make the block less internally consistent, not more (same reasoning applied to a similar finding on PR #288).

**Comment by qnbs** [comment]
@coderabbitai review
